docs/tools/near-indexer.md

```diff
@@ -0,0 +1,487 @@
+---
+id: near-indexer
+title: NEAR Indexer
+sidebar_label: NEAR Indexer
+---
+
+> The [NEAR Indexer Framework](https://github.com/near/nearcore/tree/master/chain/indexer) allows you to run a network node that listens for targeted information on the blockchain. For more information see [indexer](/docs/concepts/indexer) under our "Concepts" section.
+
+---
+
+## Setup
+
+### Requirements
+
+- [Rust.](https://www.rust-lang.org/) If not already installed, please [follow these instructions](https://docs.near.org/docs/tutorials/contracts/intro-to-rust#3-step-rust-installation).
+- Minimunm hardware:
+  - 4 CPU cores
+  - 16GB RAM
+  - 100GB SSD _(HDD will **not** work)_
+
+### Creating your project
+
+To create a new project with Rust, you will start by creating a new binary by running the following command in your terminal:
+
+```bash
+cargo new --bin example-indexer
+```
+
+Now change directory into your newly created project:
+
+```bash
+cd example-indexer
+```
+
+Inside this folder you will find:
+
+- `cargo.toml`
+- `src` folder with a `main.rs` file inside
+
+### Create Rust Toolchain
+
+Next, you will need to create a Rust toolchain that mirrors the one in [`nearcore`](https://github.com/near/nearcore/blob/master/rust-toolchain):
+
+To do this, run the following command in the root of your project: _(be sure to check the link above for the correct `nightly` version)_
+
+```bash
+echo nightly-2020-10-08 > rust-toolchain
+```
+
+### Add dependencies
+
+**1) In your `cargo.toml` file add `near-indexer` under [dependencies]:**
+
+```toml
+[dependencies]
+near-indexer = { git = "https://github.com/near/nearcore" }
+```
+
+**Note:** While it is fine to omit specific commit hash for this tutorial we highly recommend to freeze near-indexer dependency for specific commit from the `nearcore` repository. _(Example below)_
+
+```toml
+near-indexer = { git = "https://github.com/nearprotocol/nearcore", rev="29fcaf3b8c81a4c0371d105054ce251355382a77" }
+```
+
+**2) Add `actix`, `openssl-probe`, & `tokio`**
+
+```toml
+actix = "0.11.0-beta.1"
+openssl-probe = { version = "0.1.2" }
+tokio = { version = "1.1", features = ["sync"] }
+```
+
+**3) Once complete, your `cargo.toml` dependencies should look something like this:**
+
+```toml
+[dependencies]
+near-indexer = { git = "https://github.com/near/nearcore" }
+actix = "0.11.0-beta.1"
+openssl-probe = { version = "0.1.2" }
+tokio = { version = "1.1", features = ["sync"] }
+```
+
+**4) Install and check dependencies**
+
+Install and check dependencies by running:
+
+```bash
+cargo check
+```
+
+<blockquote class="warning">
```

Comment:
In the last revision I've decided to move this information to the section "Troubleshooting" to the end of the doc.

Reply:
I moved it here as I also ran into errors when running through this tutorial that these instructions fixed. There was only one "Troubeshooting" item so I moved it here to make the fix more convenient. Would you like me to move it back?